internal/sci/gcp/manager_test.go

```diff
@@ -0,0 +1,97 @@
+package gcp_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/stretchr/testify/require"
+	"github.com/substratusai/substratus/internal/sci"
+	"github.com/substratusai/substratus/internal/sci/gcp"
+	"google.golang.org/api/iam/v1"
+
+	// Can be changed to slices once we go to 1.21
+	"golang.org/x/exp/slices"
+)
+
+func TestServer(t *testing.T) {
```

Comment:
How do I call this? Will this fail if the SA was not pre-created and I run `make test`?

Reply:
you call it by running `make test` and ensuring you've set `export GOOGLE_APPLICATION_CREDENTIALS=$PWD/secrets/substratus-sa.json`

It won't fail if you have not set `GOOGLE_APPLICATION_CREDENTIALS`, it will  simply skip the test. So as soon as our CI starts loading `GOOGLE_APPLICATION_CREDENTIALS` this tests will automatically start running.